warehouse/query-core/src/main/java/datawave/query/planner/DatePartitionedQueryPlanner.java

```diff
@@ -332,29 +366,48 @@ public CloseableIterable<QueryData> process(GenericQueryConfiguration genericCon
         for (Map.Entry<Pair<Date,Date>,Set<String>> dateRange : dateRanges.entrySet()) {
+            // Get the configuration with an updated query (pushed down unindexed fields)
+            final ShardQueryConfiguration configCopy = getUpdatedConfig(planningConfig, dateRange.getKey(), dateRange.getValue());
+            futures.add(Triple.of(dateRange, configCopy, executor.submit(new Callable<CloseableIterable<QueryData>>() {
+                @Override
+                public CloseableIterable<QueryData> call() {
+                    try {
+                        // Create a copy of the original default query planner, and process the query with the new date range.
+                        DefaultQueryPlanner subPlan = DatePartitionedQueryPlanner.this.queryPlanner.clone();
+
+                        // Get the range stream for the new date range and query
+                        return subPlan.reprocess(configCopy, configCopy.getQuery(), scannerFactory);
+                    } catch (Exception e) {
+                        throw new DatawaveAsyncOperationException(e);
+                    }
+                }
+            })));
+        }
+
+        for (Triple<Map.Entry<Pair<Date,Date>,Set<String>>,ShardQueryConfiguration,Future<CloseableIterable<QueryData>>> future : futures) {
+            Map.Entry<Pair<Date,Date>,Set<String>> dateRange = future.getLeft();
             String subBeginDate = dateFormat.format(dateRange.getKey().getLeft());
             String subEndDate = dateFormat.format(dateRange.getKey().getRight());
-
-            // Get the configuration with an updated query (pushed down unindexed fields)
-            ShardQueryConfiguration configCopy = getUpdatedConfig(planningConfig, dateRange.getKey(), dateRange.getValue());
+            ShardQueryConfiguration configCopy = future.getMiddle();
 
             try {
-                // Create a copy of the original default query planner, and process the query with the new date range.
-                DefaultQueryPlanner subPlan = this.queryPlanner.clone();
-
-                // Get the range stream for the new date range and query
-                results.addIterable(subPlan.reprocess(configCopy, configCopy.getQuery(), scannerFactory));
-
+                results.addIterable(future.getRight().get());
                 if (log.isDebugEnabled()) {
                     log.debug("Query string for config of sub-plan against date range (" + subBeginDate + "-" + subEndDate + ") with unindexed fields "
                                     + dateRange.getValue() + ": " + configCopy.getQueryString());
                 }
-            } catch (DatawaveQueryException e) {
+            } catch (ExecutionException e) {
                 log.warn("Exception occurred when processing sub-plan against date range (" + subBeginDate + "-" + subEndDate + ")", e);
+                if (e.getCause() instanceof DatawaveAsyncOperationException) {
```

Comment:
Can we just leave the DatawaveAsyncOperationException and put it in the exceptions list? If we add the context I suggested above, then we'd know which piece of the query the exception came from.